framework/log/Target.php

```diff
@@ -37,7 +37,7 @@
 abstract class Target extends Component
 {
     /**
-     * @var bool whether to enable this log target. Defaults to true.
+     * @var bool|callable whether to enable this log target. Defaults to true.
      */
     public $enabled = true;
```

Comment:
It would be better to convert `enabled` to getters and setters, where the getter would evaluate the closure, so that you can still use `if ($target->enabled)` even if it is a closure. Otherwise such check would always return true instead of evaluating the closure.